src/Symfony/Component/DependencyInjection/Compiler/PassConfig.php

```diff
@@ -94,6 +94,8 @@ public function __construct()
             new ResolveHotPathPass(),
             new ResolveNoPreloadPass(),
             new AliasDeprecatedPublicServicesPass(),
+            new ResolveParameterPlaceHoldersPass(true, false),
```

Comment:
Why does it need an extra run?

Reply:
Could you illustrate with a test case? I'm not sure to understand why removing some parameters implies to resolve placeholders again, and that would prevent (re)moving this line by mistake

Reply:
Sure, this is already the case with this test https://github.com/symfony/symfony/pull/47680/files#diff-ce915f72da04411c739dd98b8350d1298450bb9258a56e6ba2d072716a15f309R76, and a parameter that could be a build parameter in the future.

Reply:
I removed this part: let it be a limitation that build-time parameters won't be resolved if they're referenced after optimization passes.